### DISCUSSION TITLE
Build failed when publishing an app with groups name

### DISCUSSION THREAD

**Discussion by Kutbi-dev** [codemagic-ci-cd/codemagic-docs#2831] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

React Native

### Steps to reproduce

Publish the build using yarn run command.
start the build process successfully but at Keychain steps it gives the error below everytime when use the external group name from codemagic setting to fetch the Key data.

GET https://api.appstoreconnect.apple.com/v1/bundleIds?limit=100&sort=name&filter%5Bidentifier%5D=com.aura.panic.quicksuresecurity.services&filter%5Bplatform%5D=IOS returned 403: A required agreement is missing or has expired. - This request requires an in-effect agreement that has not been signed or has expired.

Build failed :|
Step 5 script `Keychain` exited with status code 1

Earlier it was successfully published and generate the artefacts.  There is no change in code and repo since then.

### Expected results

Build should be successfully published and generates the artefacts.

### Actual results

Throw error below

GET https://api.appstoreconnect.apple.com/v1/bundleIds?limit=100&sort=name&filter%5Bidentifier%5D=com.aura.panic.quicksuresecurity.services&filter%5Bplatform%5D=IOS returned 403: A required agreement is missing or has expired. - This request requires an in-effect agreement that has not been signed or has expired.

Build failed :|
Step 5 script `Keychain` exited with status code 1

only when use the groups - group_name under environment variable.

### Build id (optional)

6735f660773c02454c719479

**Comment by dam-ease** [comment, marked as answer]
Hello @Kutbi-dev,
There are some agreements pending on your Apple Developer account that you'd need to confirm. Kindly visit https://appstoreconnect.apple.com/agreements/#/ and accept all the pending agreements, this issue should be resolved.
